src/popup/pages/SignMessage.vue

```diff
@@ -39,7 +39,36 @@ export default defineComponent({
         const message = isHexEncodedMessage ? Buffer.from(rawMessage, 'hex') : rawMessage;
         const displayMessage = message?.toString();
         const { host, href } = callbackOrigin.value || {} as any;
+        const jwt = route.query.jwt?.toString() === 'true';
 
+        if (jwt) {
+          const messageObject = JSON.parse(message as string);
+
+          await openModal(MODAL_MESSAGE_SIGN, {
+            message: JSON.stringify(messageObject, null, 2),
+            app: {
+              host,
+              name: host,
+              url: href,
+            },
+          });
+
+          const signature = await aeSdk.signMessageJWT(messageObject);
+          openCallbackOrGoHome(true, { signature });
+        } else {
+          await openModal(MODAL_MESSAGE_SIGN, {
+            message: displayMessage,
+            app: {
+              host,
+              name: host,
+              url: href,
+            },
+          });
+
+          const signature = await aeSdk.signMessage(message as string);
+          const signatureHex = Buffer.from(signature).toString('hex');
+          openCallbackOrGoHome(true, { signature: signatureHex });
+        }
```

Comment:
```suggestion
        } else {
```
This is actually a copy of the code underneath.

Reply:
yes it has been moved to the else case where the if is `if(jwt)`

Reply:
It wasn't moved, yet. There are currently
```
if () {
  action a;
} else {
  action b;
}
action b;
```

Reply:
oh right, just noticed, thanks